instrumentation/gwt-2.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/gwt/GwtTest.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.gwt;
+
+import static io.opentelemetry.instrumentation.testing.util.TelemetryDataUtil.orderByRootSpanName;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.equalTo;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.test.utils.PortUtils;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.concurrent.TimeUnit;
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.util.resource.Resource;
+import org.eclipse.jetty.webapp.WebAppContext;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.containers.BrowserWebDriverContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+
+class GwtTest {
+
+  @RegisterExtension
+  private static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private static final Logger logger = LoggerFactory.getLogger(GwtTest.class);
+  static int port;
+  static Server server;
+  static BrowserWebDriverContainer<?> browser;
+  static URI address;
+
+  static void startServer() throws Exception {
+    port = PortUtils.findOpenPort();
+    server = new Server(port);
+    for (Connector connector : server.getConnectors()) {
+      ((ServerConnector) connector).setHost("localhost");
+    }
+
+    WebAppContext webAppContext = new WebAppContext();
+    webAppContext.setContextPath(getContextPath());
+    webAppContext.setBaseResource(Resource.newResource(new File("build/testapp/web")));
+
+    server.setHandler(webAppContext);
+    server.start();
+  }
+
+  static void stopServer() throws Exception {
+    server.stop();
+    server.destroy();
+  }
+
+  @BeforeAll
+  static void setupSpec() throws Exception {
```

Comment:
`setupSpec` and `cleanupSpec` are spock method names. In java tests it is conventional to call these `setup` and `cleanup` instead.